SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/typeinfo/models/FixedArrayTypeInfo.h

```diff
@@ -0,0 +1,205 @@
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/helper/fixed_array.h>
+#include <sofa/defaulttype/typeinfo/DataTypeInfo.h>
+
+namespace sofa::defaulttype
+{
+
+template<class TDataType, sofa::Size static_size = TDataType::static_size>
+struct FixedArrayTypeInfo
+{
+    typedef TDataType DataType;
+
+    typedef typename DataType::value_type BaseType;
+    typedef DataTypeInfo<BaseType> BaseTypeInfo;
+    typedef typename BaseTypeInfo::ValueType ValueType;
+    typedef DataTypeInfo<ValueType> ValueTypeInfo;
+
+    enum { ValidInfo       = BaseTypeInfo::ValidInfo       };
+    enum { FixedSize       = BaseTypeInfo::FixedSize       };
+    enum { ZeroConstructor = BaseTypeInfo::ZeroConstructor };
+    enum { SimpleCopy      = BaseTypeInfo::SimpleCopy      };
+    enum { SimpleLayout    = BaseTypeInfo::SimpleLayout    };
+    enum { Integer         = BaseTypeInfo::Integer         };
+    enum { Scalar          = BaseTypeInfo::Scalar          };
+    enum { Text            = BaseTypeInfo::Text            };
+    enum { CopyOnWrite     = 1                             };
+    enum { Container       = 1                             };
+
+    enum { Size = static_size * BaseTypeInfo::Size };
+    static sofa::Size size()
+    {
+        return DataType::size() * BaseTypeInfo::size();
+    }
+
+    static sofa::Size byteSize()
+    {
+        return ValueTypeInfo::byteSize();
+    }
+
+    static sofa::Size size(const DataType& data)
+    {
+        if (FixedSize)
+            return size();
+        else
+        {
+            sofa::Size s = 0;
+            for (sofa::Size i=0; i<DataType::size(); ++i)
+                s+= BaseTypeInfo::size(data[(sofa::Size)i]);
+            return s;
+        }
+    }
+
+    static bool setSize(DataType& data, sofa::Size size)
+    {
+        if (!FixedSize)
+        {
+            size /= DataType::size();
```

Comment:
I know this is not related to this PR, but this block seems weird.

Reply:
Once the PR is merged feel free to clean the sofa code :)